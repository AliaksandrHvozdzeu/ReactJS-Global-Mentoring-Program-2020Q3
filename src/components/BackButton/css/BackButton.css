.back-button {
    font-size: 1.2rem;
    font-weight: 200;
    height: 3rem;
    padding: 0.5rem 4rem;
    color: var(--red);
    background-color: transparent;
    border: 1px solid var(--red);
    text-transform: uppercase;
    border-radius: 4px;
    outline: none;
    letter-spacing: 0.1rem;
}

.back-button:hover {
    background-color: var(--red);
    color: var(--white);
}

.back-button:active {
    box-shadow: inset 0 0 0 1px #000000;
    background-color: var(--red-darker);
}

