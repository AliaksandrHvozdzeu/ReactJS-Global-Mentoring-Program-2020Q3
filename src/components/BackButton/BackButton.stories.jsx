import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import BackButton from './js/BackButton';
import '../../App.css';

export default {
  title: 'BackButton',
  decorators: [withKnobs],
};

export const Default = () => (
  <BackButton />
);
