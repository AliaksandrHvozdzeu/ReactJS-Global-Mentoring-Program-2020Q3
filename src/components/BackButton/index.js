import BackButton from './js/BackButton';

export default BackButton;
