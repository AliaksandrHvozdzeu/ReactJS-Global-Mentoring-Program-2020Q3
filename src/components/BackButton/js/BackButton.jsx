import React from 'react';
import '../css/BackButton.css';

const BackButton = () => {

  return (
    <button type="button" className="back-button">Back</button>
  );

};

export default BackButton;
