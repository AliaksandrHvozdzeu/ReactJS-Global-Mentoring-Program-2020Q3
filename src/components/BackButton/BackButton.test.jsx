import React from 'react';
import rerender from 'react-test-renderer';
import BackButton from './js/BackButton';

describe('when BackButton', () => {
  test('then snapshot created', () => {
    const component = rerender.create(<BackButton/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
