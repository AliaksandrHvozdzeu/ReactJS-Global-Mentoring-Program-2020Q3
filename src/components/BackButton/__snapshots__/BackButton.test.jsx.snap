// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when BackButton then snapshot created 1`] = `
<button
  className="back-button"
  type="button"
>
  Back
</button>
`;
